controller/user/user_payment_controller.go

```diff
@@ -0,0 +1,105 @@
+package userpayment
+
+import (
+	"ev-payment-service/config"
+	"ev-payment-service/dao"
+	"ev-payment-service/dto"
+	helper "ev-payment-service/helper"
+	"fmt"
+	"github.com/sirupsen/logrus"
+)
+
+type UserPaymentController interface {
```

Comment:
this is the tricky portion hahah, actually we are doing the same thing. so actually payment dto is an interface. 

then extends providerPayment and userPayment. its ok no need change for now

Reply:
yes I agree, i realised this later on after userPayment is done. Lazy to refactor.